server/plugin/webhook.go

```diff
@@ -590,20 +590,87 @@ func (p *Plugin) postIssueEvent(event *github.IssuesEvent) {
 			description = *issue.Body
 		}
 
-		post := p.makeBotPost(renderedMessage, "")
+		post := &model.Post{
+			UserId: p.BotUserID,
+			Type:   "custom_git_release",
+		}
+
+		pluginURL := *p.client.Configuration.GetConfig().ServiceSettings.SiteURL + "/" + "plugins" + "/" + Manifest.Id
 
 		if action == actionOpened {
-			post.Type = "custom_git_issue"
-			post.Props = map[string]interface{}{
-				titleForProps:       *issue.Title,
-				issueURLForProps:    *issue.HTMLURL,
-				issueNumberForProps: *issue.Number,
-				descriptionForProps: description,
-				assigneesForProps:   assignees,
-				labelsForProps:      labels,
-				repoOwnerForProps:   *repo.Owner.Login,
-				repoNameForProps:    *repo.Name,
-				issueStatus:         statusClose,
+			fields := []*model.SlackAttachmentField{}
+
+			if len(assignees) > 0 {
+				fields = append(fields, &model.SlackAttachmentField{
+					Title: "Assignees",
+					Value: strings.Join(assignees, ", "),
+					Short: true,
+				})
+			}
+
+			if len(labels) > 0 {
+				fields = append(fields, &model.SlackAttachmentField{
+					Title: "Labels",
+					Value: strings.Join(labels, ", "),
+					Short: true,
+				})
+			}
+
+			post.Props = model.StringInterface{
+				"attachments": []*model.SlackAttachment{
+					{
+						Pretext:   renderedMessage,
+						Title:     fmt.Sprintf("%s #%d", *issue.Title, *issue.Number),
+						TitleLink: *issue.HTMLURL,
+						Text:      description,
+						Actions: []*model.PostAction{
+							{
+								Name: "Comment",
+								Integration: &model.PostActionIntegration{
+									Context: map[string]interface{}{
+										"repo_owner":   repo.GetOwner().GetLogin(),
+										"repo_name":    repo.GetName(),
+										"issue_number": issue.GetNumber(),
+										"issue_id":     issue.GetID(),
+										"status":       *issue.State,
+										"channel_id":   sub.ChannelID,
+									},
+									URL: fmt.Sprintf("%s/api/v1/open-comment-modal", pluginURL),
+								},
+								Style: "primary",
+							},
+							{
+								Name: "Edit",
+								Integration: &model.PostActionIntegration{
+									Context: map[string]interface{}{
+										"repo_owner":   repo.GetOwner().GetLogin(),
+										"repo_name":    repo.GetName(),
+										"issue_number": issue.GetNumber(),
+										"issue_id":     issue.GetID(),
+										"status":       *issue.State,
+										"channel_id":   sub.ChannelID,
```

Comment:
This should be present in the postActionIntegration request. So, no need to pass in context